src/io/fasta.rs

```diff
@@ -509,15 +533,18 @@ impl Record {
 
     /// Return the id of the record.
     pub fn id(&self) -> Option<&str> {
-        match self.header[1..].trim_right().splitn(2, ' ').nth(0) {
-            Some("") => None,
-            value => value,
+        match self.id.as_ref() {
+            "" => None,
```

Comment:
I know this breaks the previous API, but I think id should always return a string. There cannot be a fastq/fasta record without an id. If there is no letter behind the separator, it is simply the empty id. What do you think?